tuta-sdk/rust/src/crypto/entity_facade.rs

```diff
@@ -25,38 +25,42 @@ impl EntityFacade {
 }
 
 impl EntityFacade {
-    pub fn decrypt_and_map(&self, type_model: &TypeModel, entity: ParsedEntity, session_key: &AesKey) -> Result<ParsedEntity, ApiCallError> {
+    pub fn decrypt_and_map(&self, type_model: &TypeModel, mut entity: ParsedEntity, session_key: &AesKey) -> Result<ParsedEntity, ApiCallError> {
         let mut mapped_decrypted: HashMap<String, ElementValue> = Default::default();
-        mapped_decrypted.insert("errors".to_string(), ElementValue::Dict(Default::default()));
-        mapped_decrypted.insert("final_ivs".to_string(), ElementValue::Dict(Default::default()));
-        for (key, model_value) in type_model.values.iter().clone().into_iter() {
-            let element_value = self.map_value(entity.clone(), session_key, key, model_value);
+        let mut mapped_errors: HashMap<String, ElementValue> = Default::default();
+        let mut mapped_ivs: HashMap<String, ElementValue> = Default::default();
+
+        for (key, model_value) in type_model.values.iter() {
+            let stored_element = entity.remove_entry(key).unwrap_or_else(|| (key.to_string(), ElementValue::Null)).1;
```

Comment:
sorry for nitpick but I suggested to remove_entry to do this:

```suggestion
let (key, stored_element) = entity.remove_entry(key).unwrap_or_else(|| (key.to_string(), ElementValue::Null)).1;
```

and then you don't need to call `key.to_string()` once below

Reply:
This will save just one to_string, right? Wouldn't be better from the point of view of readability to have just one variable and use to_string for all?